plume-models/src/blocklisted_emails.rs

```diff
@@ -0,0 +1,145 @@
+use diesel::{self, ExpressionMethods, QueryDsl, RunQueryDsl, TextExpressionMethods};
+use glob::Pattern;
+
+use schema::email_blocklist;
+use {Connection, Error, Result};
+
+#[derive(Clone, Queryable, Identifiable)]
+#[table_name = "email_blocklist"]
+pub struct BlocklistedEmail {
+    pub id: i32,
+    pub email_address: String,
+    pub note: String,
+    pub notify_user: bool,
+    pub notification_text: String,
+}
+
+#[derive(Insertable, FromForm)]
+#[table_name = "email_blocklist"]
+pub struct NewBlocklistedEmail {
+    pub email_address: String,
+    pub note: String,
+    pub notify_user: bool,
+    pub notification_text: String,
+}
+
+impl BlocklistedEmail {
+    insert!(email_blocklist, NewBlocklistedEmail);
+    get!(email_blocklist);
+    find_by!(email_blocklist, find_by_id, id as i32);
+    pub fn delete_entries(conn: &Connection, ids: Vec<i32>) -> Result<bool> {
+        use diesel::delete;
+        for i in ids {
+            let be: BlocklistedEmail = BlocklistedEmail::find_by_id(&conn, i)?;
+            delete(&be).execute(conn)?;
+        }
+        Ok(true)
+    }
+    pub fn find_for_domain(conn: &Connection, domain: &str) -> Result<Vec<BlocklistedEmail>> {
+        let effective = format!("%{}", domain);
```

Comment:
Maybe add a `@` so that in only matches one domain, and not all domains with the same ending (I doubt this case will ever be encountered actually, be just to be sure).